src/drv_oculus_rift/rift-hmd-radio.c

```diff
@@ -0,0 +1,295 @@
+	  uint16_t json_offset = offset - 4;
+
+		ret = rift_radio_read_flash(handle, device_type, offset, 20, flash_data);
+		if (ret < 0) {
+			free(json);
+			return ret;
+		}
+
+		memcpy(json + json_offset, flash_data, OHMD_MIN (20, json_length - json_offset));
+	}
+
+	*json_out = json;
+	*length = json_length;
+
+	return 0;
+}
+
+static bool
+json_read_vec3 (const nx_json *nxj, const char *key, vec3f *out)
+{
+	const nx_json *member = nx_json_get (nxj, key);
+
+	if (member->type != NX_JSON_ARRAY)
+		return false;
+
+	out->x = nx_json_item (member, 0)->dbl_value;
+	out->y = nx_json_item (member, 1)->dbl_value;
+	out->z = nx_json_item (member, 2)->dbl_value;
+
+	return true;
+}
+
+static int rift_touch_parse_calibration(
+					char *json,
+					rift_touch_calibration *c)
+{
+	const nx_json* nxj, *obj, *version, *array;
+	int version_number = -1;
+	unsigned int i;
+
+	nxj = nx_json_parse (json, 0);
+	if (nxj == NULL)
+		return -1;
+
+	obj = nx_json_get(nxj, "TrackedObject");
+	if (obj->type == NX_JSON_NULL)
+		goto fail;
+
+	version = nx_json_get (obj, "JsonVersion");
+	if (version->type != NX_JSON_INTEGER || version->int_value != 2) {
+		version_number = version->int_value;
+		goto fail;
+	}
+	version_number = version->int_value;
+
+  if (!json_read_vec3 (obj, "ImuPosition", &c->imu_position))
+		goto fail;
+
+	c->joy_x_range_min = nx_json_get (obj, "JoyXRangeMin")->int_value;
+	c->joy_x_range_max = nx_json_get (obj, "JoyXRangeMax")->int_value;
+	c->joy_x_dead_min = nx_json_get (obj, "JoyXDeadMin")->int_value;
+	c->joy_x_dead_max = nx_json_get (obj, "JoyXDeadMax")->int_value;
+	c->joy_y_range_min = nx_json_get (obj, "JoyYRangeMin")->int_value;
+	c->joy_y_range_max = nx_json_get (obj, "JoyYRangeMax")->int_value;
+	c->joy_y_dead_min = nx_json_get (obj, "JoyYDeadMin")->int_value;
+	c->joy_y_dead_max = nx_json_get (obj, "JoyYDeadMax")->int_value;
+
+	c->trigger_min_range = nx_json_get (obj, "TriggerMinRange")->int_value;
+	c->trigger_mid_range = nx_json_get (obj, "TriggerMidRange")->int_value;
+	c->trigger_max_range = nx_json_get (obj, "TriggerMaxRange")->int_value;
+
+	array = nx_json_get (obj, "GyroCalibration");
+	for (i = 0; i < 12; i++)
+		c->gyro_calibration[i] = nx_json_item (array, i)->dbl_value;
+
+	c->middle_min_range = nx_json_get (obj, "MiddleMinRange")->int_value;
+	c->middle_mid_range = nx_json_get (obj, "MiddleMidRange")->int_value;
+	c->middle_max_range = nx_json_get (obj, "MiddleMaxRange")->int_value;
+
+	c->middle_flipped = nx_json_get (obj, "MiddleFlipped")->int_value;
+
+	array = nx_json_get (obj, "AccCalibration");
+	for (i = 0; i < 12; i++)
+		c->acc_calibration[i] = nx_json_item (array, i)->dbl_value;
+
+	array = nx_json_get (obj, "CapSenseMin");
+	for (i = 0; i < 8; i++)
+		c->cap_sense_min[i] = nx_json_item (array, i)->int_value;
+
+	array = nx_json_get (obj, "CapSenseTouch");
+	for (i = 0; i < 8; i++)
+		c->cap_sense_touch[i] = nx_json_item (array, i)->int_value;
+
+	/* LED model points */
+#if 0
```

Comment:
Please don't add dead code. I'd prefer this to be enabled even if it isn't hooked up yet, but feel free to drop it for now.

Reply:
I'll move it to a patch on the sensor branch.